test/controllers/test_widgets_controller.rb

```diff
@@ -0,0 +1,26 @@
+require 'test_helper'
+
+class Devise::WidgetsControllerTest < ActionController::TestCase
+  include Devise::Test::ControllerHelpers
+
+  setup do
+    @controller = WidgetsController.new
+    @request.env['devise.mapping'] = Devise.mappings[:user]
+    @user = User.create!(
+      username: 'hello',
+      email: 'hello@path.travel',
+      password: 'Password4',
+      password_changed_at: 4.months.ago,
+      confirmed_at: 5.months.ago
+    )
+    assert @user.valid?
+    assert @user.need_change_password?
+
+    sign_in(@user)
+  end
+
+  test 'should render show' do
```

Comment:
Was this test added to help debug, or was there something it helped validate? 

I don't recall why the user wouldn't be redirected to `user_password_expired_path` like in the other test.  I tried poking around and adding an `assert_redirected_to`, and confirmed it was **not** redirecting.

Reply:
This was added to try and track down the problem, but this may now be superseded by other tests. It doesn't redirect because `WidgetsController` does not inherit from `Devise::PasswordExpiredController`